ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client/HistoricityCheck.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DerivingStrategies  #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving  #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Ouroboros.Consensus.MiniProtocol.ChainSync.Client.HistoricityCheck (
+    -- * Interface
+    HistoricalChainSyncMessage (..)
+  , HistoricityCheck (..)
+  , HistoricityCutoff (..)
+  , HistoricityException (..)
+    -- * Real implementation
+  , mkCheck
+  , noCheck
+  ) where
+
+import           Control.Exception (Exception)
+import           Control.Monad (when)
+import           Control.Monad.Except (throwError)
+import           Data.Time.Clock (NominalDiffTime)
+import           Data.Typeable (eqT)
+import           Ouroboros.Consensus.Block
+import           Ouroboros.Consensus.BlockchainTime (RelativeTime,
+                     SystemTime (..), diffRelTime)
+import           Ouroboros.Consensus.HeaderStateHistory
+                     (HeaderStateWithTime (..))
+import           Ouroboros.Consensus.HeaderValidation (HasAnnTip,
+                     headerStatePoint)
+import           Ouroboros.Consensus.Node.GsmState (GsmState (..))
+
+{-------------------------------------------------------------------------------
+  Interface
+-------------------------------------------------------------------------------}
+
+-- | Interface for the ChainSync client for deciding whether @MsgRollBackward@s
+-- and @MsgAwaitReply@s are historical.
+data HistoricityCheck m blk = HistoricityCheck {
+    -- | Determine whether the received @MsgRollBackward@ is historical. The
```

Comment:
This comment seems stale in a couple ways, not that MsgAwaitReply is also involved.

Reply:
Thanks, changed to
```haskell
    -- | Determine whether the received message is historical. Depending on the
    -- first argument, the second argument is:
    --
    --  * 'HistoricalMsgRollBackward': The oldest state that was rolled back.
    --    (Note that rollbacks of depth zero are hence never historical.).
    --
    --  * 'HistoricalMsgAwaitReply': The state corresponding to the tip of the
    --    candidate fragment when @MsgAwaitReply@ was sent.
```